packages/react-native-builder-bob/babel-config.js

```diff
@@ -61,7 +61,16 @@ const getConfig = (defaultConfig, { root, pkg }) => {
       },
       {
         include: path.join(root, src),
-        presets: [require.resolve('./babel-preset')],
+        presets: [
+          [
+            require.resolve('./babel-preset'),
+            {
+              // Let React Native preset handle the commonjs transform
+              // Otherwise this causes issues with `@react-native/babel-plugin-codegen`
+              supportsStaticESM: true,
```

Comment:
@jbroma codegen generates `export` in invalid position after the file is transformed to commonjs

Reply:
the CJS transform comes from `@babel/preset-env` and codegen runs as a plugin and the order of execution should be plugins -> presets, this shouldnt be a problem 🤔

Reply:
the codegen plugin runs as part of the react native preset and presets run in the reverse order iirc, which means codegen plugin will run after preset-env

Reply:
> the codegen plugin runs as part of the react native preset and presets run in the reverse order iirc, which means codegen plugin will run after preset-env

but we don't use the RN preset here? and codegen plugin shouldn't run twice since we strip the types here

Reply:
@jbroma it is used in `...defaultConfig` where example app passes the RN preset. codegen only runs as part of the RN preset.